File: Second Nanodegree/Organização e Abstração na Programação/Atividades/python/Veiculo.py
class Veiculo:
    def __init__(self, marca, modelo, placa, ano):
        self.__marca = marca
        self.__modelo = modelo
        self.__placa = placa
        self.__ano = ano

    # Métodos getters and setters para os atributos privados
    def get_marca(self):
        return self.__marca
    def set_marca(self, marca):
        self.__marca = marca

    def get_modelo(self):
        return self.__modelo
    def set_modelo(self, modelo):
        self.__modelo = modelo

    def get_placa(self):
        return self.__placa
    def set_placa(self, placa):
        self.__placa = placa

    def get_ano(self):
        return self.__ano
    def set_ano(self, ano):
        self.__ano = ano


    def calcularTempoUso(self):
        return 2024 - self.__ano
    ## altera a formatação padrao da string da classe
    def __str__(self):
        return f'''- Modelo: {self.__modelo}
- Marca: {self.__marca}
- Placa: {self.__placa} 
- Ano: {self.__ano}'''

File: Second Nanodegree/Organização e Abstração na Programação/Atividades/java/Veiculo.java
public class Veiculo {
    //atributos
    private String placa;
    private int ano;
    private String marca;
    
    public Veiculo() {
    }
    
    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public int getAno() {
        return ano;
    }

    public void setAno(int ano) {
        this.ano = ano;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public Veiculo(String placa, int ano) {
        this.placa = placa;
        this.ano = ano;
    }

    public Veiculo cloneMe(){
        //clonar as prop nesse novo veiculo e mandar de volta
        return new Veiculo();
    }
}